From e619280ec1d0c7c3258a2dcae3b360b194122c96 Mon Sep 17 00:00:00 2001
From: Johannes Raggam <raggam-nl@adm.at>
Date: Sat, 30 Oct 2010 13:16:39 +0100
Subject: [PATCH] Use tom_gross's improved and tested saveToInt function

---
 Products/CMFPlone/utils.py |   28 ++++++++++++++++++++++++----
 1 files changed, 24 insertions(+), 4 deletions(-)

diff --git a/Products/CMFPlone/utils.py b/Products/CMFPlone/utils.py
index 74343d5..2f08643 100644
--- a/Products/CMFPlone/utils.py
+++ b/Products/CMFPlone/utils.py
@@ -306,12 +306,32 @@ def _createObjectByType(type_name, container, id, *args, **kw):
     return fti._constructInstance(container, id, *args, **kw)
 
 
-def safeToInt(value):
-    """Convert value to integer or just return 0 if we can't"""
+def safeToInt(value, default=0):
+    """Convert value to integer or just return 0 if we can't
+
+       >>> safeToInt(45)
+       45
+
+       >>> safeToInt("42")
+       42
+
+       >>> safeToInt("spam")
+       0
+
+       >>> safeToInt([])
+       0
+
+       >>> safeToInt(None)
+       0
+
+       >>> safeToInt(None, default=-1)
+       -1
+    """
     try:
         return int(value)
-    except ValueError:
-        return 0
+    except (ValueError, TypeError):
+        return default
+
 
 release_levels = ('alpha', 'beta', 'candidate', 'final')
 rl_abbr = {'a':'alpha', 'b':'beta', 'rc':'candidate'}
-- 
1.7.1

