Index: plips/plipbase.cfg
===================================================================
--- plips/plipbase.cfg	(revision 29663)
+++ plips/plipbase.cfg	(working copy)
@@ -8,5 +8,35 @@
 parts-directory = ../parts
 sources-dir = ../src
 
+parts +=
+    zexps
+    coreloadtests
+develop +=
+    ../../collective.loadtesting/src/collective.coreloadtests
+
 [instance]
 var = ${buildout:directory}/../var
+debug-mode = off
+eggs +=
+    collective.contentgenerator
+    collective.coreloadtests    
+zcml +=
+    collective.contentgenerator
+    collective.coreloadtests    
+
+[zexps]
+recipe = iw.recipe.cmd
+on_update = True
+on_install = True
+cmds = ln -s ${instance:var}/instance/public1.zexp ${instance:var}/instance/public2.zexp ${instance:var}/instance/public3.zexp ${instance:var}/instance/public4.zexp ${instance:var}/instance/public5.zexp ${instance:location}/import/
+
+[coreloadtests]
+recipe = collective.recipe.plonesite >= 0.6
+site-id = coreloadtests
+site-replace = True
+profiles-initial = 
+    profile-collective.coreloadtests:writeheavy 
+    profile-collective.coreloadtests:contentcreation
+    profile-collective.contentgenerator:public
+zexps-cmds = ${zexps:cmds}
+
